controllers/profile.js

```diff
@@ -14,7 +14,7 @@ exports.getProfile = function (req, res) {
 };
 
 exports.updateProfile = function (req, res) {
-    return Users.findOneAndUpdate(req.user._id, req.body.user, {
+    return Users.findByIdAndUpdate(req.user._id, req.body.user, {

```

Comment:
From the [mongoose docs](https://mongoosejs.com/docs/api.html#model_Model.findByIdAndUpdate), it looks like `findByIdAndUpdate(id, ...)` is equivalent to `findOneAndUpdate({ _id: id }, ...)`. The implicit conversion of the ID field to the filter JSON `{_id: id}` seems to be working on my machine, but not yours. Can we switch this to be `findOneAndUpdate` with the explicit filter object, since `findByIdAndUpdate` ends up calling `findOneAndUpdate`, and I'd prefer we do the call directly.

Reply:
thanks for pulling up the docs, just to clarify the issue:
The docs state that `findByIdAndUpdate(id, ...)` is equivalent to `findOneAndUpdate({ _id: id }, ...)`, but in our case we were using `findOneAndUpdate(id, ...)` which is not equivalent to `findByIdAndUpdate(id, ...)`

I changed it to `findOneAndUpdate` as requested